src/main/java/com/patchworkmc/event/EventSubscriptionChecker.java

```diff
@@ -0,0 +1,70 @@
+package com.patchworkmc.event;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+// Currently patchwork cannot handle @SubscribeEvent for overloaded methods
+public class EventSubscriptionChecker {
+	private static class Entry {
+		public List<SubscribeEvent> subscribeEvents;
+		public List<String> superClasses;
+
+		Entry(List<SubscribeEvent> subscribeEvents, List<String> superClasses) {
+			this.subscribeEvents = subscribeEvents;
+			this.superClasses = superClasses;
+		}
+	}
+
+	private Map<String, Entry> data = new HashMap<>();
+
+	public EventSubscriptionChecker() {
+	}
+
+	public void onClassScanned(
+			String className,
+			List<SubscribeEvent> subscribeEvents,
+			List<String> superClasses
+	) {
+		data.put(className, new Entry(subscribeEvents, superClasses));
+	}
+
+	public void check() {
+		data.forEach((className, entry) -> {
+			Set<String> descriptions = new HashSet<>();
+			gatherSuperClasses(className).distinct().flatMap(
+					c -> data.get(c).subscribeEvents.stream()
+			).forEach(subscribeEvent -> {
+				String description = subscribeEvent.getMethod()
+						+ subscribeEvent.getEventClass()
+						+ subscribeEvent.getGenericClass().orElse(null);
+				if (descriptions.contains(description)) {
+					throw new RuntimeException(
+							String.format(
+									"Currently Patchwork cannot handle @SubscribeEvent for overloaded methods. %s %s",
+									className,
+									subscribeEvent
+							)
+					);
+				} else {
+					descriptions.add(description);
+				}
+			});
+		});
+	}
+
+	private Stream<String> gatherSuperClasses(String className) {
+		return Optional.ofNullable(data.get(className))
+				.map(entry -> Stream.concat(
+						entry.superClasses.stream().flatMap(
+								this::gatherSuperClasses
+						),
+						Stream.of(className)
+				))
+				.orElse(Stream.empty());
+	}
```

Comment:
I think this would be better implemented by adding entries to a HashSet recursively similar to `EventBus#typesFor`. It's a bit hard to see what the code is doing as is with the use of functional programming. Also, I think that missing superclass information, with the exception of `java.*`, `net.minecraft.*`, and `net.minecraftforge.*` (and maybe an expanded whitelist based on libraries), should result in a hard error for now. An unknown superclass *could* contain subscribe events, and without superclass data, it's not possible to verify for sure.

Finally, since all this is used for is collecting a list of `SubscribeEvent` entries, instead of returning the names of the classes it would be just as easy to return a list of all the `SubscribeEvent` entries instead.